src/Sylius/Bundle/ApiBundle/Applicator/StateMachineTransitionApplicator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Applicator;
+
+use SM\Factory\FactoryInterface as StateMachineFactoryInterface;
+use Sylius\Component\Core\Model\ProductReview;
+use Sylius\Component\Core\ProductReviewTransitions;
+use Sylius\Component\Review\Model\ReviewInterface;
+
+final class StateMachineTransitionApplicator
```

Comment:
```suggestion
final class ProductReviewStateMachineTransitionApplicator
```

Also, if we want to keep this class this way, I would introduce an interface, or split this class into separate applicators.